### DISCUSSION TITLE
Use a library for command line processing

### DISCUSSION THREAD

**Discussion by Thihup** [piranhacloud/piranha#1588] [Ideas]
One way of improving the usage of Piranha would be by having a good CLI. Currently, we're doing all the argument processing by hand, which is repetitive and error-prone.

There is the [Picocli](https://github.com/remkop/picocli) library which seems to be easy to use, can be compiled to native image, and does not have transitive dependencies.

Its documentation seems to be pretty good: https://picocli.info

Now we have three applications that would benefit from it: Piranha CLI, Piranha Micro, and Piranha Server

**Comment by manorrock** [comment]
Go for it!
